pandas/tests/test_categorical.py

```diff
@@ -803,6 +803,12 @@ def f():
             cat.remove_categories(["c"])
         self.assertRaises(ValueError, f)
 
+        # Remove a NaN category with underlying float type (GH #10156)
+        result = Categorical([], categories=[1.0, 2.0, np.nan])
+        result = result.remove_categories(np.nan)
+        expected = Categorical([], categories=[1.0, 2.0])
+        self.assert_categorical_equal(result, expected)
+
     def test_remove_unused_categories(self):
```

Comment:
can you add a test for this as well ``>>> pd.Categorical([], categories=[np.nan, None])``
also add in using ``None`` in the ``.remove_categories``. Pls test on an object dtype as well as floating-point.
Bonus points if you can make this work for datetimelike (e.g. using ``pd.NaT``.